x-pack/plugins/enterprise_search/server/lib/enterprise_search_config_api.ts

```diff
@@ -67,44 +67,60 @@ export const callEnterpriseSearchConfigAPI = async ({
+          id: data?.current_user?.app_search?.role?.id,
+          roleType: data?.current_user?.app_search?.role?.roleType,
           ability: {
-            accessAllEngines: !!data?.current_user?.app_search_role?.ability?.access_all_engines,
-            destroy: data?.current_user?.app_search_role?.ability?.destroy || [],
-            manage: data?.current_user?.app_search_role?.ability?.manage || [],
-            edit: data?.current_user?.app_search_role?.ability?.edit || [],
-            view: data?.current_user?.app_search_role?.ability?.view || [],
-            credentialTypes: data?.current_user?.app_search_role?.ability?.credential_types || [],
+            accessAllEngines: !!data?.current_user?.app_search?.role?.ability?.access_all_engines,
+            destroy: data?.current_user?.app_search?.role?.ability?.destroy || [],
+            manage: data?.current_user?.app_search?.role?.ability?.manage || [],
+            edit: data?.current_user?.app_search?.role?.ability?.edit || [],
+            view: data?.current_user?.app_search?.role?.ability?.view || [],
+            credentialTypes: data?.current_user?.app_search?.role?.ability?.credential_types || [],
             availableRoleTypes:
-              data?.current_user?.app_search_role?.ability?.available_role_types || [],
+              data?.current_user?.app_search?.role?.ability?.available_role_types || [],
           },
         },
       },
       workplaceSearch: {
-        canCreateInvitations: !!data?.settings?.workplace_search?.can_create_invitations,
-        isFederatedAuth: !!data?.settings?.workplace_search?.is_federated_auth,
         organization: {
-          name: data?.settings?.workplace_search?.organization?.name,
-          defaultOrgName: data?.settings?.workplace_search?.organization?.default_org_name,
+          name: data?.current_user?.workplace_search?.organization?.name,
+          defaultOrgName: data?.current_user?.workplace_search?.organization?.default_org_name,
         },
-        fpAccount: {
-          id: data?.settings?.workplace_search?.fp_account.id,
-          groups: data?.settings?.workplace_search?.fp_account.groups || [],
-          isAdmin: !!data?.settings?.workplace_search?.fp_account?.is_admin,
-          canCreatePersonalSources: !!data?.settings?.workplace_search?.fp_account
+        account: {
+          id: data?.current_user?.workplace_search?.account?.id,
+          groups: data?.current_user?.workplace_search?.account?.groups || [],
+          isAdmin: !!data?.current_user?.workplace_search?.account?.is_admin,
+          canCreatePersonalSources: !!data?.current_user?.workplace_search?.account
             ?.can_create_personal_sources,
-          isCurated: !!data?.settings?.workplace_search?.fp_account.is_curated,
-          viewedOnboardingPage: !!data?.settings?.workplace_search?.fp_account
-            .viewed_onboarding_page,
+          canCreateInvitations: !!data?.current_user?.workplace_search?.account
```

Comment:
@scottybollinger FYI - `canCreateInvitations` moved from being a top-level WS key to nested under `account`